worlds/aquaria/__init__.py

```diff
@@ -0,0 +1,216 @@
+"""
+Author: Louis M
+Date: Fri, 15 Mar 2024 18:41:40 +0000
+Description: Main module for Aquaria game multiworld randomizer
+"""
+
+from typing import List, Dict, ClassVar, Any
+from ..AutoWorld import World, WebWorld
+from BaseClasses import Tutorial, MultiWorld, ItemClassification, LocationProgressType
+from .Items import item_table, AquariaItem, ItemType, ItemGroup
+from .Locations import location_table
+from .Options import AquariaOptions
+from .Regions import AquariaRegions
+
+
+class AquariaWeb(WebWorld):
+    """
+    Class used to generate the Aquaria Game Web pages (setup, tutorial, etc.)
+    """
+    theme = "ocean"
+
+    bug_report_page = "https://github.com/tioui/Aquaria_Randomizer/issues"
+
+    setup = Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up Aquaria for MultiWorld.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["Tioui"]
+    )
+
+    setup_fr = Tutorial(
+        "Guide de configuration Multimonde",
+        "Un guide pour configurer Aquaria MultiWorld",
+        "Français",
+        "setup_fr.md",
+        "setup/fr",
+        ["Tioui"]
+    )
+
+    tutorials = [setup, setup_fr]
+
+
+class AquariaWorld(World):
+    """
+    Aquaria is a side-scrolling action-adventure game. It follows Naija, an
+    aquatic humanoid woman, as she explores the underwater world of Aquaria.
+    Along her journey, she learns about the history of the world she inhabits
+    as well as her own past. The gameplay focuses on a combination of swimming,
+    singing, and combat, through which Naija can interact with the world. Her
+    songs can move items, affect plants and animals, and change her physical
+    appearance into other forms that have different abilities, like firing
+    projectiles at hostile creatures, or passing through barriers inaccessible
+    to her in her natural form.
+    From: https://en.wikipedia.org/wiki/Aquaria_(video_game)
+    """
+
+    game: str = "Aquaria"
+    "The name of the game"
+
+    topology_present = True
+    "show path to required location checks in spoiler"
+
+    web: WebWorld = AquariaWeb()
+    "The web page generation informations"
```

Comment:
```suggestion
    game = "Aquaria"
    "The name of the game"

    topology_present = True
    "show path to required location checks in spoiler"

    web = AquariaWeb()
    "The web page generation informations"
```

No need to type these since that'd be an override rather than just setting the instance variable (according to pycharm at least)

Reply:
Well. Here, I disagree. Personally, I think the IDE is useful, but the code should be self explanatory and readable on it's own. And the typing system of python (that is not perfect of course, but still very useful) can make it more easy to read the code. So, if someone want to read the code directly on github or with a text editor (other than PyCharm), the reading will be harder without the typing.